File: Pattern_Strategy/app/src/main/java/com/example/cnwlc/pattern_strategy/robot/MainActivity.java
package com.example.cnwlc.pattern_strategy.robot;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.cnwlc.pattern_strategy.InterFace.FlyYes;
import com.example.cnwlc.pattern_strategy.InterFace.KnifeLaser;
import com.example.cnwlc.pattern_strategy.InterFace.MisailYes;
import com.example.cnwlc.pattern_strategy.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        superRobot();
    }
    private void superRobot() {
        System.out.println("SuperRobot을 만들어 주세요.");

        Robot superRobot = new SuperRobot();
        superRobot.shape();
        superRobot.actionRun();
        superRobot.actionWalk();

//        superRobot.setFly(new FlyYes());
//        superRobot.setMisail(new MisailYes());
//        superRobot.setKnife(new KnifeLaser());
        // set 을 잊어먹거나 하기 귀찮으면 SuperRobot 안에 디폴트 값을 주고 변경하고 싶으면 위 방식처럼 new FlyNo() 로 바꿔주면 된다.
        // default 값을 안줄거면 위처럼 그냥 할당해주면 됨.

        superRobot.actionFly();
        superRobot.actionMisail();
        superRobot.actionKnife();
    }
}

File: Test_weather/app/src/main/java/restaurant/study/com/test_weather/MainActivity.kt
package restaurant.study.com.test_weather

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by jc_chu on 2018. 08. 05..
 */

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        main_bt.setOnClickListener {
            startActivity(Intent(this, GpsActivity::class.java))
        }
    }
}
